rclpy/rclpy/utilities.py

```diff
@@ -71,9 +72,37 @@ def get_rmw_implementation_identifier():
 
 
 def get_available_rmw_implementations():
-    """Return the set of all available RMW implementations as registered in the ament index."""
-    rmw_implementations = ament_index_python.get_resources('rmw_typesupport')
-    return {name for name in rmw_implementations if name != 'rmw_implementation'}
+    """
+    Return the set of all available RMW implementations as registered in the ament index.
+
+    The result can be overridden by setting an environment variable named
+    ``RMW_IMPLEMENTATIONS``.
+    The variable can contain RMW implementation names separated by the platform
+    specific path separator.
+    Including an unavailable RMW implementation results in a RuntimeError.
+    """
+    available_rmw_implementations = ament_index_python.get_resources(
+        'rmw_typesupport')
+    available_rmw_implementations = {
+        name for name in available_rmw_implementations
+        if name != 'rmw_implementation'}
+
+    # filter by implementations in environment variable if provided
+    rmw_implementations = os.environ.get('RMW_IMPLEMENTATIONS')
```

Comment:
Suggest renaming the variable to `filtered_rmw_implementations`.

The more that I think about it, the more I feel like we should rename the environment variable. `RMW_IMPLEMENTATIONS` is very similar to the existing environment variable `RMW_IMPLEMENTATION`.

What if we changed the environment variable name to `FILTER_AVAILABLE_RMM_IMPLEMENTATIONS`? I know it's a little verbose.

Reply:
Imo both variables are similar since they do similar things. `RMW_IMPLEMENTATION` selects the one to be used, `RMW_IMPLEMENTATIONS` lists the ones which can be used. Adding `FILTER_` to one sounds arbitrary - the same argument could be made for prefixing the other with `SELECT_` / `CHOOSE_`.

Reply:
Thought I'd raise the concern; I'm not going to block on it.

Reply:
If we want to we can still change it. As long as we haven't released the change a modification in a follow up PR should be straight forward.